ndcube/global_coords/global_coords.py

```diff
@@ -0,0 +1,83 @@
+from collections import OrderedDict
+from collections.abc import Mapping
+
+__all__ = ['GlobalCoords']
+
+
+class GlobalCoords(Mapping):
+    """
+    A structured representation of coordinate information applicable to a whole NDCube.
+    """
+    def __init__(self, ndcube):
+        """
+        Init method.
+        """
+        super().__init__()
+        self._ndcube = ndcube
+
+        # Set values using the setters for validation
+        self._internal_coords = OrderedDict()
+
+    @property
+    def _all_coords(self):
+        """
+        Establish the _all_coords property with an _internal_coords.
+        """
+        return self._internal_coords
+
+    def add(self, name, physical_type, coords):
+        """
+        Add a new coordinate to the collection.
+        """
+        self._internal_coords[str(name)] = (name, physical_type, coords)
```

Comment:
I still think we shouldn't have name in the value of this internal dict. I can see no use for it when it's already the key. Also I don't think we need to cast to `str()` for the key, unless we are going to type check all the input.

```suggestion
        self._internal_coords[name] = (physical_type, coords)
```